utils/bin/poller/src/main.rs

```diff
@@ -0,0 +1,301 @@
+use cow_amm_common::{
+    rpc_url, ComposableCoW,
+    ConstantProductHelper::{self, ConstantProductHelperErrors},
+    GPv2Settlement,
+    IConstantProductHelper::{self, dReturn},
+    IMulticall3, IPriceOracle, LegacyConstantProduct, LegacyTradingParams, Order, IERC1271,
+};
+use std::collections::HashMap;
+
+use std::str::FromStr;
+
+use alloy::{
+    contract::Error,
+    hex,
+    primitives::{Address, Bytes, FixedBytes, U256},
+    providers::{Provider, ProviderBuilder},
+    rpc::types::state::AccountOverride,
+    sol_types::{
+        decode_revert_reason, eip712_domain, SolCall, SolEvent, SolInterface, SolStruct, SolValue,
+    },
+    transports::TransportError,
+};
+
+#[tokio::main]
+async fn main() -> eyre::Result<()> {
+    const SETTLEMENT: &str = "0x9008D19f58AAbD9eD0D60971565AA8510560ab41";
+    const MUTLICALL3: &str = "0xcA11bde05977b3631167028862bE2a173976CA11";
+
+    // Create a provider.
+    let provider = ProviderBuilder::new().on_http(rpc_url());
+    let chain_id = provider.get_chain_id().await?;
+
+    // Which amm are we interested in?
+    let amm: Address = std::env::var("AMM")
+        .expect("Environment variable `AMM` is not set")
+        .parse()
+        .expect("Invalid address specified for AMM");
+
+    println!("Polling AMM: {:?}", amm);
+
+    // First configure the helper and the overrides
+    const HELPER: &str = "0xBEEF5AFeBEef5aFeBEeF5AfEBEef5AfEBEef5AFE";
+    let helper = ConstantProductHelper::new(HELPER.parse()?, provider.clone());
+    let mut overrides = HashMap::new();
+    overrides.insert(
+        HELPER.parse()?,
+        AccountOverride {
+            code: Some(ConstantProductHelper::DEPLOYED_BYTECODE.clone()),
+            ..Default::default()
+        },
+    );
+
+    // Second get the raw data that comes from the snapshot within the ConstantProductHelper
+    // This is needed so that we can lookup an oracle, unless we otherwise have the price to
+    // supply.
+    let data = helper
+        .getSnapshot(amm)
+        .state(overrides.clone())
+        .call()
+        .await?;
+
+    // Check if the AMM is legacy
+    let legacy = !data._0.is_empty();
+
+    let numerator = std::env::var("NUMERATOR");
+    let denominator = std::env::var("DENOMINATOR");
+
+    let (numerator, denominator, other_post_interactions) = match legacy {
```

Comment:
Very deep into the review I was (wrongly) convinced that `other_post_interactions` was just about setting the commit after the fact and not just reading the commit. The confusion originated from the fact that this is _not_ intended to be a post interaction, it's just a call to retrieve the output of `commitment` after the fact.
My suggested name is `call_commitment_read`, also I wouldn't set it in this match block because it's orthogonal to setting the price and can just be done independently without making this already heavy block heavier. There should also be an explanatory comment.